packages/react-topology/src/components/TopologyView/examples/TopologyControlBarDemo.tsx

```diff
@@ -0,0 +1,202 @@
+      </g>
+    </DefaultNode>
+  );
+};
+
+const customLayoutFactory: LayoutFactory = (type: string, graph: Graph): Layout | undefined => {
+  switch (type) {
+    case 'Cola':
+      return new ColaLayout(graph);
+    default:
+      return new ColaLayout(graph, { layoutOnDrag: false });
+  }
+};
+
+const customComponentFactory: ComponentFactory = (kind: ModelKind, type: string) => {
+  switch (type) {
+    case 'group':
+      return DefaultGroup;
+    default:
+      switch (kind) {
+        case ModelKind.graph:
+          return withPanZoom()(GraphComponent);
+        case ModelKind.node:
+          return withSelection()(CustomNode);
+        case ModelKind.edge:
+          return withSelection()(DefaultEdge);
+        default:
+          return undefined;
+      }
+  }
+};
+
+const NODE_DIAMETER = 75;
+
+const NODES: NodeModel[] = [
+  {
+    id: 'node-0',
+    type: 'node',
+    label: 'Node 0',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER
+  },
+  {
+    id: 'node-1',
+    type: 'node',
+    label: 'Node 1',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER
+  },
+  {
+    id: 'node-2',
+    type: 'node',
+    label: 'Node 2',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER
+  },
+  {
+    id: 'node-3',
+    type: 'node',
+    label: 'Node 3',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER
+  },
+  {
+    id: 'node-4',
+    type: 'node',
+    label: 'Node 4',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER
+  },
+  {
+    id: 'node-5',
+    type: 'node',
+    label: 'Node 5',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.rect
+  },
+  {
+    id: 'Group-1',
+    children: ['node-0', 'node-1', 'node-2'],
+    type: 'group',
+    group: true,
+    label: 'Group-1',
+    style: {
+      padding: 40
+    }
+  }
+];
+
+const EDGES = [
+  {
+    id: 'edge-node-4-node-5',
+    type: 'edge',
+    source: 'node-4',
+    target: 'node-5',
+    edgeStyle: EdgeStyle.default
+  },
+  {
+    id: 'edge-node-0-node-2',
+    type: 'edge',
+    source: 'node-0',
+    target: 'node-2',
+    edgeStyle: EdgeStyle.default
+  }
+];
+
+export const TopologySelectableDemo: React.FC = () => {
+  const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
+
+  const controller = React.useMemo(() => {
+    const model: Model = {
+      nodes: NODES,
+      edges: EDGES,
+      graph: {
+        id: 'g1',
+        type: 'graph',
+        layout: 'Cola'
+      }
+    };
+
+    const newController = new Visualization();
+    newController.registerLayoutFactory(customLayoutFactory);
+    newController.registerComponentFactory(customComponentFactory);
+
+    newController.addEventListener(SELECTION_EVENT, setSelectedIds);
+
+    newController.fromModel(model, false);
+
+    return newController;
+  }, []);
+
+  return (
+    <TopologyView
+      controlBar={
+        <TopologyControlBar
+          controlButtons={createTopologyControlButtons({
```

Comment:
We should explain this parameter above, what the`defaultControlButtonsOptions`  are and how to override the defaults.